### PR TITLE
Fix compilation without notification

### PR DESCRIPTION
**Pull request by vingu-linaro**
Hi All,

As discussed, I'm splitting PR #524  in smaller ones. This PR is a subset of it which contains the compilation error

### CODE REVIEW

**Comment by tarek-arm** [comment]
Jenkins: OK to test

File: module/clock/src/mod_clock.c [hunk 1]
```diff
@@ -29,7 +29,6 @@ static struct clock_ctx mod_clock_ctx;
 /*
  * Utility functions
  */
-
```
Comment (etienne-lms) [inline module/clock/src/mod_clock.c:32]: maybe unrelated change

File: module/power_domain/src/mod_power_domain.c [hunk 2]
```diff
@@ -1402,6 +1415,9 @@ static bool check_and_notify_system_shutdown(
     }
 
     return (mod_pd_ctx.system_shutdown.notifications_count != 0);
+#else
+    return 0;
```
Comment (etienne-lms) [inline module/power_domain/src/mod_power_domain.c:1419]: s/0/false/

**Review by etienne-lms — COMMENTED** [review]
LGTM with some minor comments.
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` once addressed (for what it's worth).

Could you also pull the equivalent fix on _module/reset_domain/_ (https://github.com/etienne-lms/SCP-firmware/commit/26410b5a77a96ed30437723d23e91988012fa1c1) in this series? or do I create another P-R?

**Comment by vingu-linaro** [comment]
> LGTM with some minor comments. `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` once addressed (for what it's worth).
> 
> Could you also pull the equivalent fix on _module/reset_domain/_ ([etienne-lms@26410b5](https://github.com/etienne-lms/SCP-firmware/commit/26410b5a77a96ed30437723d23e91988012fa1c1)) in this series? or do I create another P-R?

Thanks

I can add reset_domain changes in this PR

**Comment by vingu-linaro** [comment]
Hi All, I have pushed a new version with a additional patch for reset_domain

File: module/dvfs/src/mod_dvfs.c [hunk 3]
```diff
@@ -1127,13 +1127,17 @@ static int dvfs_bind_element(fwk_id_t domain_id, unsigned int round)
 
     /* Bind to the alarm HAL if required */
     if (ctx->config->retry_ms > 0) {
```
Comment (tarek-arm) [inline module/dvfs/src/mod_dvfs.c:1129]: If the timer is not used you can set config->retry_ms = 0 this will skip using the timer.

**Review by tarek-arm — CHANGES_REQUESTED** [review]
I think we can drop this patch as it's not required.

File: module/dvfs/src/mod_dvfs.c (hunk 3 above)
Comment (vingu-linaro) [reply module/dvfs/src/mod_dvfs.c:1129]: when compiling without timer, FWK_MODULE_IDX_TIMER is undefined and as a result MOD_TIMER_API_ID_ALARM is undefined too

Comment (tarek-arm) [reply module/dvfs/src/mod_dvfs.c:1129]: You have 2 options:
1. Include the timer module but not use it.
2. #if BUILD_HAS_MOD_TIMER all of the timer related structures, variables and functions in dvfs module. This includes the config structure in dvfs header file.

I am not sure how this patch compiles if you are not including the timer.

Comment (tarek-arm) [reply module/dvfs/src/mod_dvfs.c:1129]: I don't know what build environment you are using but this will not work for our build system. Even your last patch is not enough.
I think the easiest way for you just to include the timer module without using it. We have examples in the code base like this[ product config file](https://github.com/ARM-software/SCP-firmware/blob/master/product/rdn2/scp_romfw/config_clock.c) The clock is included to gain access to the header file. This is option 1.
OR if you want NOT to include the timer then you need to #if all the timer related strcut,func,var in the dvfs .c and .h files

**Comment by tarek-arm** [comment]
Jenkins: Ok to test

**Comment by tarek-arm** [comment]
@vingu-linaro we have just completed v2.10 release. Can you please rebase this PR and push an update so we can merge it?
Everything else is fine by me.

**Comment by vingu-linaro** [comment]
> @vingu-linaro we have just completed v2.10 release. Can you please rebase this PR and push an update so we can merge it? Everything else is fine by me.

@tarek-arm, I have rebased and pushed this PR on master branch.
Thanks

File: module/reset_domain/src/mod_reset_domain.c [hunk 4]
```diff
@@ -105,8 +107,12 @@ static int rd_process_event(
                          event->id))
         return FWK_E_SUPPORT;
```
[the remark this answers is not in the document]
Comment (vingu-linaro) [reply module/reset_domain/src/mod_reset_domain.c:111]: > Sorry, testing this change reports a build warning on unused `params` variable in the function. @vingu-linaro, could you squash this following [fixup commit](https://github.com/etienne-lms/SCP-firmware/commit/2db4bcf58c7c431549d7c812d2cca1e68c5b5077) into the last commit in this series?
> 
> ```diff
>  {
> +#ifdef BUILD_HAS_NOTIFICATION
>      struct mod_reset_domain_autoreset_event_params* params =
>          (struct mod_reset_domain_autoreset_event_params*)event->params;
> +#endif
> ```

I have updated the patchset with the change above

**Comment by tarek-arm** [comment]
Jenkins: OK to test

**Review by tarek-arm — APPROVED** [review]

**Comment by vingu-linaro** [comment]
Thanks @tarek-arm . I'm going to prepare next patchset
